pkg/disk/partition_table.go

```diff
@@ -1083,6 +1083,20 @@ func addBootPartition(pt *PartitionTable, bootFsType FSType) error {
 	return nil
 }
 
+func hasESP(disk *blueprint.DiskCustomization) bool {
```

Comment:
This seems like a good candidate for a struct method on DiskCustomization since it only depends on info from the struct in question.

Reply:
I'm not sure its really worth making such a small helper be part of the disk customization object directly. Its quite specific. Having it as a regular function is similar to e.g. needsBoot() in the same file.